template.go

```diff
@@ -167,51 +182,50 @@ func (t *Template) Execute(r Recaller) (*ExecuteResult, error) {
 		return nil, errors.Wrap(err, "execute")
 	}
 
+	// Checks if all values in use have been fetched.
+	// ..also cleans out data no longer used by this template.
+	if !w.Complete(t) {
+		return nil, ErrMissingValues
+	}
+
 	return &ExecuteResult{
-		Used:    *used,
-		Missing: *missing,
-		Output:  b.Bytes(),
+		Output: b.Bytes(),
```

Comment:
I thought about that as I was cleaning out the other stuff but was in the middle of things and wasn't sure. I'll make this change and see how it looks.

Reply:
Required changes were spread out a bit, but not to bad. I'll push it up along with the rest (soon).

Reply:
Here's the commit with them... I think this is good to go, but for context and in case.

https://github.com/hashicorp/hcat/pull/26/commits/9842aa03006bb6e422e0f073f527cc2ca827dc3e